crates/epaint/src/text/text_layout.rs

```diff
@@ -560,30 +560,55 @@ fn halign_and_justify_row(
         return;
     }
 
-    let num_leading_spaces = row
-        .glyphs
-        .iter()
-        .take_while(|glyph| glyph.chr.is_whitespace())
-        .count();
-
-    let glyph_range = if num_leading_spaces == row.glyphs.len() {
-        // There is only whitespace
-        (0, row.glyphs.len())
```

Comment:
I kept this logic in my refactor below. And I see that it avoids possible underflow when/after we get to the `assert!()` but I'm still a bit fuzzy on why we even continue in that case.  Do we need to shift all of the spaces into a left/center/right-aligned location? So... what, they're in a useful place when selecting label texts?